sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/NestedColumnAliasing.scala

```diff
@@ -348,6 +382,11 @@ object GeneratorNestedColumnAliasing {
         case _ =>
       }
 
+      g.generator match {
+        case _: ExplodeBase =>
+        case _ => return Some(pushedThrough)
+      }
```

Comment:
I think you can just remove `Inline` from `canPruneGenerator`.

Reply:
Discussed below.